lib/delegate.rb

```diff
@@ -136,6 +136,7 @@ def ==(obj)
     return true if obj.equal?(self)
     self.__getobj__ == obj
   end
+  alias :eql? :==
```

Comment:
Sorry, I didn't see this, my mail filtering was broken.
Looks good, the trick to peel with `__getobj__` and then call `eql?` on `obj` is nice,
particularly for user classes which just test for a specific class like below.
It allows two Delegator (even nested) to compare properly.
I think `Delegator#==` should use the same trick.
```ruby
[12] pry(main)> class T; def ==(o); T === o; end; end # Just to illustrate, should compare fields as well  
[13] pry(main)> a=T.new
[14] pry(main)> b=T.new
[15] pry(main)> a==b
=> true
[16] pry(main)> SimpleDelegator.new(a) == b
=> true
[17] pry(main)> SimpleDelegator.new(a) == SimpleDelegator.new(b)
=> false
[18] pry(main)> class T; def eql?(o); T===o; end; end
[19] pry(main)> SimpleDelegator.new(a) == SimpleDelegator.new(b)
=> false
[20] pry(main)> SimpleDelegator.new(a).eql? SimpleDelegator.new(b)
=> true
```